src/pages/DetailsPage.js

```diff
@@ -123,7 +122,7 @@ function DetailsPage(props) {
     const phoneNumber = getPhoneNumber(details);
     const phoneOrEmail = isSMSLogin ? getPhoneNumber(details) : details.login;
 
-    const isCurrentUser = _.keys(props.loginList).includes(details.login);
+    const isCurrentUser = props.session.email === login.toLowerCase();
```

Comment:
We should use the account id.

Reply:
But we can get accountID here only by checking corresponding login in personal details.
When we login with deeplink, personal details are not loaded immediately so account id is undefined and the check returns false even for same current user and the message button appears briefly.

Reply:
Hmm in that case please revert. We can't compare with session.email because one account can have multiple emails.

Reply:
Or wait, currently we are comparing details.login from onyx and not login from route params. So if we just use details.accountID then we won't be introducing a new change (in terms of deep link behavior)

Reply:
Yea but that does not help in fixing this bug either.
If we get directly from URL then we can get the current user check instantly. But like you said, the primary login may be different so session.email === email from url does not work.
Changing to accountID also has no use in this case. Loginlist and personaldetails are loaded more or less at the same time.

Reply:
I understand that the change to DetailsPage will fix nothing. but also will break nothing right? I'm suggesting this just for consistency i.e. to have the isCurrentUser always related to the account id

Reply:
Got it. I will make that change.

Reply:
Done